main/influxdb_watcher.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2017 Hewlett Packard Enterprise Development LP
+//
+//    Licensed under the Apache License, Version 2.0 (the "License"); you may
+//    not use this file except in compliance with the License. You may obtain
+//    a copy of the License at
+//
+//         http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+//    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+//    License for the specific language governing permissions and limitations
+//    under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	configEnv "github.com/caarlos0/env"
+	"github.com/influxdata/influxdb/client/v2"
+	"github.com/monasca/monasca-watchers/watcher"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	log "github.com/sirupsen/logrus"
+)
+
+type watcherConfiguration struct {
+	InfluxdbAddress    string `env:"INFLUXDB_ADDRESS" envDefault:"http://influxdb:8086"`
+	Username           string `env:"INFLUXDB_USERNAME" envDefault:"influxdb_watcher"`
+	Password           string `env:"INFLUXDB_PASSWORD" envDefault:"password"`
+	PrometheusEndpoint string `env:"PROMETHEUS_ENDPOINT" envDefault:"0.0.0.0:8080"`
+	Period             int64  `env:"WATCHER_PERIOD" envDefault:"600"`
+	Timeout            int64  `env:"WATCHER_TIMEOUT" envDefault:"60"`
+}
+
+// InfluxdbBroker Watcher for InfluxDB
+type InfluxdbBroker struct {
+	MonascaDB  string
+	Connection client.Client
+}
+
+// Message the values Watcher is expecting to send and receive
+type Message struct {
+	UUID     string
+	SentTime string
+}
+
+// WriteMessage Write a point to Influxdb with UUID and SentTime
+func (broker *InfluxdbBroker) WriteMessage(byteMessage []byte) error {
+	var message Message
+	err := json.Unmarshal(byteMessage, &message)
```

Comment:
This is really cheating. The message is supposed to be opaque. Is there no option to just write some bytes to influx?

Reply:
done